### PR TITLE
[varlen attn] batch invariance testing

### PR DESCRIPTION
**Pull request by liangel-02**
Tests batch invariance for varlen attention - permutations of samples in the batch should not affect fwd output and backward gradients

### CODE REVIEW

File: test/test_varlen_attention.py [hunk 1]
```diff
@@ -428,6 +436,89 @@ def test_varlen_vs_sdpa(self, device, dtype, is_causal):
 
             start_idx = end_idx
 
+    @skipIfRocm(msg="ROCM does not support variable length attention")
+    @unittest.skipIf(
+        not PLATFORM_SUPPORTS_FLASH_ATTENTION, "Flash Attention not supported"
+    )
+    @parametrize("dtype", [torch.bfloat16, torch.float16])
+    @parametrize("is_causal", [False, True])
+    def test_batch_invariance(self, device, dtype, is_causal):
```
Comment (drisspg) [inline test/test_varlen_attention.py]: lets maybe simplifiy it for this and not do a full attention (in projs and out projs) and just call the varlen func and lets use torch.testing.make_tensor since it produces higher variance data

File: test/test_varlen_attention.py [hunk 2]
```diff
@@ -428,6 +436,143 @@ def test_varlen_vs_sdpa(self, device, dtype, is_causal):
...
+            orig_end = cu_seq_orig[orig_idx + 1].item()
+            orig_seq_output = original_output[orig_start:orig_end]
+
+            perm_start = cu_seq_perm[i].item()
+            perm_end = cu_seq_perm[i + 1].item()
+            perm_seq_output = permuted_output[perm_start:perm_end]
+
+            self.assertEqual(orig_seq_output, perm_seq_output)
+
+        original_grad_out = torch.ones_like(original_output)
```
Comment (drisspg) [inline test/test_varlen_attention.py]: randn

File: test/test_varlen_attention.py [hunk 3]
```diff
@@ -428,6 +436,143 @@ def test_varlen_vs_sdpa(self, device, dtype, is_causal):
...
+        permuted_grad_out = torch.ones_like(permuted_output)
+
+        original_grad = torch.autograd.grad(
+            outputs=original_output,
+            inputs=q_packed_orig,
+            grad_outputs=original_grad_out,
+        )[0]
```
Comment (drisspg) [inline test/test_varlen_attention.py]: test all the grads don tjust test grad_q

File: test/test_varlen_attention.py [hunk 4]
```diff
@@ -428,6 +436,143 @@ def test_varlen_vs_sdpa(self, device, dtype, is_causal):
 
             start_idx = end_idx
 
+    @skipIfRocm(msg="ROCM does not support variable length attention")
+    @unittest.skipIf(
+        not PLATFORM_SUPPORTS_FLASH_ATTENTION, "Flash Attention not supported"
+    )
+    @parametrize("dtype", [torch.bfloat16, torch.float16])
+    @parametrize("is_causal", [False, True])
```
Comment (drisspg) [inline test/test_varlen_attention.py:444]: paraemtize the num perms to test

**Review by drisspg — APPROVED** [review]
